powerauth-java-server/src/main/java/com/wultra/security/powerauth/app/server/service/behavior/tasks/ActivationServiceBehavior.java

```
@@ -683,10 +663,6 @@ public GetActivationStatusResponse getActivationStatus(GetActivationStatusReques
                 response.setVersion(0L);
                 return response;
             }
-        } catch (InvalidKeySpecException | InvalidKeyException ex) {
-            logger.error(ex.getMessage(), ex);
-            // Rollback is not required, database is not used for writing
-            throw localizationProvider.buildExceptionForCode(ServiceError.INVALID_KEY_FORMAT);
```

The exception `InvalidKeySpecException` is now handled differently in the `CryptographyServiceEc256`.
Just pointing out that some specific errors are swallowed to a generic one in this PR on many places.

---

Yes, that is OK. These cryptography errors occur only when the cryptography fails, and we have sufficient logging for it and we do not need to propagate the error specifics to the client.

---

At any case I will review the individual cases so we match the original error handling as much as possible, even I don't think we should be too specific for these errors.